apps/nextjs/src/hooks/useLessonPlanScrollManagement.ts

```diff
@@ -0,0 +1,155 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { LessonPlanKeys } from "@oakai/aila/src/protocol/schema";
+import { aiLogger } from "@oakai/logger";
+
+const log = aiLogger("lessons");
+
+export const useLessonPlanScrollManagement = (
+  streamingSection: LessonPlanKeys | undefined,
+  streamingSectionCompleted: LessonPlanKeys | undefined,
+) => {
+  const [recentlyScrolledSection, setRecentlyScrolledSection] = useState<
+    LessonPlanKeys | undefined
+  >(undefined);
+
+  const [userHasCancelledAutoScroll, setUserHasCancelledAutoScroll] =
+    useState(false);
+
+  const sectionRefs = useRef<
+    Record<LessonPlanKeys, React.MutableRefObject<HTMLDivElement | null>>
+  >({
+    title: { current: null },
+    subject: { current: null },
+    keyStage: { current: null },
+    topic: { current: null },
+    learningOutcome: { current: null },
+    learningCycles: { current: null },
+    priorKnowledge: { current: null },
+    keyLearningPoints: { current: null },
+    misconceptions: { current: null },
+    keywords: { current: null },
+    basedOn: { current: null },
+    starterQuiz: { current: null },
+    exitQuiz: { current: null },
+    cycle1: { current: null },
+    cycle2: { current: null },
+    cycle3: { current: null },
+    additionalMaterials: { current: null },
+  });
+
+  const setSectionRef = useCallback(
+    (section: LessonPlanKeys, el: HTMLDivElement | null) => {
+      sectionRefs.current[section].current = el;
+    },
+    [],
+  );
+
+  const scrollQueueRef = useRef<LessonPlanKeys[]>([]);
+  const activeScrollRef = useRef<LessonPlanKeys | null>(null);
+
+  const processScrollQueue = useCallback(() => {
```

Comment:
Adds queueing behaviour so that during streaming we don't end up with scrolling thrash